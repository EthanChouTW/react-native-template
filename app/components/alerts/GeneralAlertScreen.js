import React from 'react';
import PropTypes from 'prop-types';
import { StyleSheet, Text, View, TouchableOpacity, KeyboardAvoidingView } from 'react-native';
import Theme from '../../utils/styleCollection';
import Locales from '../../locales';

const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: 'flex-start',
    alignItems: 'stretch',
    backgroundColor: Theme.hbTransparentBlack
  },
  alertContainer: {
    marginHorizontal: 30,
    padding: 15,
    height: 150,
    borderRadius: 5,
    backgroundColor: Theme.whiteColor
  },
  itemTitle: {
    flex: 1,
    justifyContent: 'center'
  },
  itemTitleText: {
    fontWeight: 'bold',
    fontSize: 16
  },
  actionButtons: {
    flex: 1,
    flexDirection: 'row',
    justifyContent: 'flex-end',
    alignItems: 'center'
  },
  actionButton: {
    marginTop: 5,
    padding: 5,
    paddingRight: 10,
    alignItems: 'flex-end'
  },
  actionButtonText: {
    color: Theme.algaeGreenColor,
    fontWeight: 'bold'
  }
});

const GeneralAlertComponent = props => (
  <KeyboardAvoidingView style={styles.container} behavior="padding">
    <View style={{ flex: 1 }} />
    <View style={styles.alertContainer}>
      <View style={styles.itemTitle}>
        <Text style={styles.itemTitleText}>{props.title}</Text>
      </View>
      <View style={styles.actionButtons}>
        <TouchableOpacity
          style={styles.actionButton}
          onPress={props.onDismissView}
        >
          <Text style={styles.actionButtonText}>{Locales.t('lbl_ok').toUpperCase()}</Text>
        </TouchableOpacity>
      </View>
    </View>
    <View style={{ flex: 1 }} />
  </KeyboardAvoidingView>
);

GeneralAlertComponent.propTypes = {
  onDismissView: PropTypes.func,
  title: PropTypes.string
};

export default GeneralAlertComponent;
