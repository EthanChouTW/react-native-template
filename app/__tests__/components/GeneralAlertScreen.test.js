import 'react-native';
import React from 'react';
// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';
import GeneralAlertScreen from '../../components/alerts/GeneralAlertScreen';

describe('General alert', () => {
  const props = {
    onDismissView: jest.fn(),
    title: 'enen'
  };
  test('renders correctly', () => {
    const snapshot = renderer.create(<GeneralAlertScreen {...props} />).toJSON();
    expect(snapshot).toMatchSnapshot();
  });
});
