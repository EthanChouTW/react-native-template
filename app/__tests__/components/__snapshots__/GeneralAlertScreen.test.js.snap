// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`General alert renders correctly 1`] = `
<
  keyboardVerticalOffset={0}
  onLayout={[Function]}
  style={
    Array [
      Object {
        "alignItems": "stretch",
        "backgroundColor": "rgba(0, 0, 0, 0.9)",
        "flex": 1,
        "justifyContent": "flex-start",
      },
      Object {
        "paddingBottom": 0,
      },
    ]
  }
>
  <
    style={
      Object {
        "flex": 1,
      }
    }
  />
  <
    style={
      Object {
        "backgroundColor": "#fff",
        "borderRadius": 5,
        "height": 150,
        "marginHorizontal": 30,
        "padding": 15,
      }
    }
  >
    <
      style={
        Object {
          "flex": 1,
          "justifyContent": "center",
        }
      }
    >
      <
        accessible={true}
        allowFontScaling={true}
        disabled={false}
        ellipsizeMode="tail"
        style={
          Object {
            "fontSize": 16,
            "fontWeight": "bold",
          }
        }
      >
        enen
      </>
    </>
    <
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "flexDirection": "row",
          "justifyContent": "flex-end",
        }
      }
    >
      <
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        collapsable={undefined}
        hitSlop={undefined}
        isTVSelectable={true}
        nativeID={undefined}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "flex-end",
            "marginTop": 5,
            "opacity": 1,
            "padding": 5,
            "paddingRight": 10,
          }
        }
        testID={undefined}
        tvParallaxProperties={undefined}
      >
        <
          accessible={true}
          allowFontScaling={true}
          disabled={false}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "rgba(43, 214, 101, 1)",
              "fontWeight": "bold",
            }
          }
        >
          LBL_OK
        </>
      </>
    </>
  </>
  <
    style={
      Object {
        "flex": 1,
      }
    }
  />
</>
`;
